polaris.shopify.com/pages/api/service/index.tsx

```diff
@@ -0,0 +1,94 @@
+import type {NextApiResponse, NextApiRequest} from 'next';
+
+const isProd = process.env.NODE_ENV === 'production';
+const enableLogs = process.env.DEBUG_ANALYTICS === 'true';
+const ANALYTICS_URL = isProd
+  ? 'https://polaris.sfe.shopifyinternal.com/analytics'
+  : 'http://localhost:3001/analytics'; // local service
+
+type Event = {
+  id: string;
+  payload: any;
+};
+
+type AnalyticsProducerFunction = (event: Event) => void;
+
+type AnalyticsProducerType = 'log' | 'http';
+
+class AnalyticsProducer {
+  constructor(producer: AnalyticsProducerType) {
+    this.produce =
+      producer === 'http' ? this.produceHttpRequest : this.produceLog;
+  }
+
+  public produce: AnalyticsProducerFunction;
+
+  private produceLog(event: Event) {
+    if (!isProd && enableLogs) {
+      console.log('Analytics event logged', event);
+    }
+  }
+
+  private async produceHttpRequest(event: Event) {
+    if (isProd) {
+      try {
+        const response = await fetch(ANALYTICS_URL, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(event),
+        });
+        const data = await response.json();
+
+        if (response.status !== 200 || data.error) {
+          logError(`Analytics call failed. Error: ${data.error}`);
```

Comment:
Might want to log the status here as well if data.error is null

```suggestion
          logError(`Analytics call failed. Error: ${data.error}. Status: ${response.status}`);
```